src/main/java/com/tfgm/models/TramStop.java

```diff
@@ -1,82 +1,231 @@
+
+  /**
+   * Constructor for class TramStop.
+   *
+   * @param stopName Name of the stop.
+   * @param direction Direction of stop.
+   * @param line Line of the stop.
+   */
+  public TramStop(String stopName, String direction, String line) {
     this.stopName = stopName;
-    this.line = line;
     this.direction = direction;
-    this.id = id;
+    this.line = line;
   }
 
-  public TramStop(TramStop tramStop) {
-    this.stopName = tramStop.getStopName();
-    this.line = tramStop.getLine();
-    this.direction = tramStop.getDirection();
-    this.line = tramStop.getLine();
-    this.id = tramStop.getId();
-    this.tramTrips = tramStop.getTramTrips();
+  /**
+   * A method to set the previous and next stops. These are in the {@link
+   * com.tfgm.models.TramStopContainer} form as they require a "link".
+   *
+   * @param prevStops The previous stops.
+   * @param nextStops The next stops.
+   */
+  public void setPrevAndNextStops(TramStopContainer[] prevStops, TramStopContainer[] nextStops) {
+    this.prevStops = prevStops;
+    this.nextStops = nextStops;
   }
 
-  public boolean addTramTrip(TramTrip tramTrip) {
-    return tramTrips.add(tramTrip);
+  /**
+   * An override of the toString method that avoids infinite loops.
+   *
+   * @return A string version of the class.
+   */
+  @Override
+  public String toString() {
+    return "NewTramStop [stopName="
+        + stopName
+        + ", direction="
+        + direction
+        + ", line="
+        + line
+        + ", nextStops: "
+        + (nextStops.length > 0
+            ? Arrays.stream(nextStops).map(n -> n.getTramStop().getStopName()).toList()
+            : "none")
+        + ", prevStops: "
+        + (prevStops.length > 0
+            ? Arrays.stream(prevStops).map(n -> n.getTramStop().getStopName()).toList()
+            : "none")
+        + "]";
   }
 
-  public String getStopName() {
-    return stopName;
+  /**
+   * Simple nextStops getter.
+   *
+   * @return Returns the nextStops array
+   */
+  public TramStopContainer[] getNextStops() {
+    return nextStops;
   }
 
-  public void setStopName(String stopName) {
-    this.stopName = stopName;
+  /**
+   * Simple prevStops getter.
+   *
+   * @return Returns the prevStops array
+   */
+  public TramStopContainer[] getPrevStops() {
+    return prevStops;
   }
 
-  public String getDirection() {
-    return direction;
+
+  public String getStopName() {
+    return stopName;
   }
 
-  public void setDirection(String direction) {
-    this.direction = direction;
+  public String getDirection() {
+    return direction;
   }
 
-  public int getId() {
-    return id;
+  public ArrayList<String> getLastUpdated() {
+    return lastUpdated;
   }
 
-  public void setId(int id) {
-    this.id = id;
+  public void addToLastUpdated(String lastUpdated) {
+    this.lastUpdated.add(lastUpdated);
   }
 
-  public String getLine() {
-    return line;
+  public void tramDeparture(String endOfLine) {
```

Comment:
As discussed, try to keep data manipulations in the data layer, the model should serve as a representation of the data but not mutate it.